### PR TITLE
Clean compile, break-fix v1

### PR DESCRIPTION
**Pull request by ProgrammerDan**
@WildWeazel  NetherFactory needs to recompute scaling factor during repair, so NetherFactoryManager is a hard dependency. Simply fixed compilation error by pulling the manager from the plugin. Not tested locally, that's up next, but IRL first. I'd be much appreciative if you could pull and test locally.

### CODE REVIEW

**Comment by rourke750** [comment]
ok to test

**Comment by rourke750** [comment]
test this please

**Comment by rourke750** [comment]
ok to test

**Comment by rourke750** [comment]
I think its fixed :/

**Comment by ProgrammerDan** [comment]
Plop it on civtest and I'll run the paces

**Comment by rourke750** [comment]
@ttk2 You dont need to merge a pull to put it in testing.  Just grab that jar that was made in the pull request bot.

**Comment by rourke750** [comment]
@ttk2 We should start being consistent in only merging pulls after testing is completed.

**Comment by ProgrammerDan** [comment]
@rourke750 full agreement, harder to back out the merge then not do it in the first place :)

**Comment by ttk2** [comment]
ok then, will do, its on Civtest and seems to fix the stack though, hop over there and try

**Comment by ProgrammerDan** [comment]
no startup exceptions at all? that's great news if so

**Comment by ttk2** [comment]
none that i saw, let me try again.

**Comment by ttk2** [comment]
2015-06-15 22:22:47 [INFO] [FactoryMod] Enabling FactoryMod v1.4.0
2015-06-15 22:22:47 [INFO] FactoryMod v1.4.0: Initializing FactoryMod Config.
2015-06-15 22:22:47 [INFO] FactoryMod v1.4.0: Finished initializing FactoryMod Config.
2015-06-15 22:22:47 [INFO] FactoryMod v1.4.0: Initiaiting FactoryMod Managers.
2015-06-15 22:22:47 [INFO] FactoryMod v1.4.0: Finished initializing FactoryMod Managers.

**Comment by ProgrammerDan** [comment]
Pretty sparse startup messages, but looks good. About to spam factory creation, hold your hat.

Also, @ttk2 noticed a general chat message of "NCP: Your configuration might be outdated. Some settings have changed, you should regenerate it!"

Not sure if that matters for civtest.

**Comment by ttk2** [comment]
ncp does that about every day, when I do change it I have to spend weeks
tweaking block break times again.

**Comment by ProgrammerDan** [comment]
ahh
So repair factory built just fine, but there was no message to chat on successful creation.

**Comment by ProgrammerDan** [comment]
@ttk2 can you check test server logs right now? Compactor isn't installing, config for it is in config.yml and its fully involved in the code, but 64 dblocks isn't producing the factory.

**Comment by ttk2** [comment]
2015-06-15 22:45:08 [INFO] FactoryMod v1.4.0: Factory creation attempted:
23068d2b-9421-44a4-8db5-745ff921bad9 at world (4319 61 -4557)
2015-06-15 22:45:08 [ERROR] Could not pass event PlayerInteractEvent to
FactoryMod v1.4.0
org.bukkit.event.EventException
        at
org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:305)
~[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62)
~[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
org.bukkit.craftbukkit.v1_8_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:227)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
org.bukkit.craftbukkit.v1_8_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:194)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.PlayerInteractManager.a(PlayerInteractManager.java:107)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.PlayerConnection.a(PlayerConnection.java:604)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.PacketPlayInBlockDig.a(SourceFile:40)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.PacketPlayInBlockDig.a(SourceFile:10)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.PlayerConnectionUtils$1.run(SourceFile:13)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[?:1.8.0_45-internal]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[?:1.8.0_45-internal]
        at net.minecraft.server.v1_8_R2.SystemUtils.a(SourceFile:60)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.MinecraftServer.A(MinecraftServer.java:710)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.DedicatedServer.A(DedicatedServer.java:368)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.MinecraftServer.z(MinecraftServer.java:651)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at
net.minecraft.server.v1_8_R2.MinecraftServer.run(MinecraftServer.java:554)
[spigot.jar:git-Spigot-1d14d5f-7722428]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45-internal]
Caused by: java.lang.NullPointerException
        at
com.github.igotyou.FactoryMod.managers.FactoryManagerService.createFactory(FactoryManagerService.java:204)
~[?:?]
        at
com.github.igotyou.FactoryMod.listeners.FactoryModListener.createFactory(FactoryModListener.java:467)
~[?:?]
        at
com.github.igotyou.FactoryMod.listeners.FactoryModListener.createFactory(FactoryModListener.java:440)
~[?:?]
        at
com.github.igotyou.FactoryMod.listeners.FactoryModListener.playerInteractionEvent(FactoryModListener.java:219)
~[?:?]
        at sun.reflect.GeneratedMethodAccessor71.invoke(Unknown Source)
~[?:?]
        at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
~[?:1.8.0_45-internal]
        at java.lang.reflect.Method.invoke(Method.java:497)
~[?:1.8.0_45-internal]
        at
org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301)
~[spigot.jar:git-Spigot-1d14d5f-7722428]
        ... 18 more

**Comment by ProgrammerDan** [comment]
Mossy Cobblestone can be crafted again.

Checkout other FactoryMod restrictions.

**Comment by WildWeazel** [comment]
Alright, this is clearly not ready for multiplayer testing yet. It doesn't look like anyone tested their respective branches first. Let me get the VM set up so I can run Citadel so I can run FM (ugh) and I'll work out the obvious blockers. @ttk2 Is there anything useful to be tested with the version from before the IO/compactor merge?

**Comment by ProgrammerDan** [comment]
Emerald Blocks -> Emeralds -> Bottles and back works

None of the special shaped crafting recipes work.

**Comment by ttk2** [comment]
I don't think we had any changes before the merge.

**Comment by ProgrammerDan** [comment]
good news, regular factories seem to be working, I'll throw up a NF real quick.

**Comment by ProgrammerDan** [comment]
So, I wrote some command blocks to produce lored elements; no luck on the NF. 

Any stack traces from the last 15 minutes?

**Comment by ProgrammerDan** [comment]
Repair factory appears to successfully reset costs.

**Comment by ttk2** [comment]
none except that one I already sent you.

**Comment by ProgrammerDan** [comment]
Alright, I'll poke at NF tonight locally. Might be I'm doing something wrong in construction; I verified that the Command Block produced ingredients matched the Aspect Factory outputs, so the input requirements should have been met, put the coal & obby block above the chest ... but no output, no error, no message back to the player.
